### PR TITLE
Remove openvino_builders

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Remove openvino_builders cmake target
 - Move all builders to test utils or to dev_api

### Tickets:
 - CVS-127030

### CODE REVIEW

**Comment by olpipi** [comment]
@vurusovs I suggest to wait with merge until 2023.3 goes to separate branch

File: src/core/tests/CMakeLists.txt [hunk 1]
```diff
@@ -38,9 +38,9 @@ ov_add_test_target(
         LINK_LIBRARIES
             common_test_utils
             openvino::reference
-            openvino::builders
             openvino::util
             openvino::shape_inference
+            ngraph_obj
```
Comment (ilya-lavrenov) [inline src/core/tests/CMakeLists.txt:43]: looks like it should not be required

File: src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp
Comment (ilya-lavrenov) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp:1]: why these node builders are not part of https://github.com/openvinotoolkit/openvino/tree/0d5aae2a09b48ac6cc4bfc358ffbadd3e4e17fb2/src/tests/ov_helpers/ov_models/include/ov_models ?

Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp:1]: Some of them are used in production code. So I moved them to dev api.
Builders which are used only in tests were moved to tests utils

File: src/core/tests/CMakeLists.txt (hunk 1 above)
Comment (olpipi) [reply src/core/tests/CMakeLists.txt:43]: It is. Tests use these functions. I moved the functions to core part. So now core tests requires static linkage to core lib.

File: src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp
Comment (ilya-lavrenov) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp:1]: Utils are generic things, while specific builders are closer to ov_models IMO

@vurusovs your opinion?

File: src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp [hunk 2]
```diff
@@ -4,14 +4,15 @@
...
-namespace ngraph {
-namespace builder {
+#include "openvino/op/constant.hpp"
+#include "openvino/core/type/float16.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
 template <class T>
 std::shared_ptr<Node> make_constant(const element::Type& type, const Shape& shape, const T& num) {
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp:17]: Is it used somewhere?

File: src/core/dev_api/openvino/op/builder/reshape.hpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace ov {
+namespace utils {
+/// \brief      Change shape of a value
+///
+/// \param[in]  value  The value to be reshaped.
+/// \param[in]  shape  The new shape.
+///
+/// \return     Reshape:v1 op.
+std::shared_ptr<Node> reshape(const Output<Node>& value, const Shape& shape);
```
Comment (vurusovs) [inline src/core/dev_api/openvino/op/builder/reshape.hpp:17]: let's use same format as `ov::op::util::make_slice_plan`

Comment (vurusovs) [reply src/core/dev_api/openvino/op/builder/reshape.hpp:17]: in this case same folders layout may be used

File: src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp
Comment (vurusovs) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/node_builders/constant.hpp:1]: Let's discuss offline and find proper solution here

**Comment by olpipi** [comment]
@ilya-lavrenov  @vurusovs  please take a look again

**Review by gkrivor — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]
I request to move `tests/ov_helpers/ov_models/ov_builders` to `src\core\dev_api\openvino\op\util` in order to not have `openvino::builders` at all and not to add dependency on `tests` folder

It should be discussed with @ilya-lavrenov

**Pull request merged**